piff/optical_model.py

```diff
@@ -0,0 +1,203 @@
+
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: optical_model
+"""
+
+from __future__ import print_function
+
+import galsim
+import fitsio
+import numpy
+
+from .model import Model
+from .stardata import StarData
+from .starfit import Star, StarFit
+
+class Optical(Model):
+    def __init__(self, rzero=0.1, sigma=0., g1=0., g2=0., pupil_path='', lam=500., **kwargs):
+        """Initialize the Optical Model
+
+        :param rzero:       Atmospheric seeing. Usually in the 0.1 - 0.2 range.
+        :param g1, g2:      Shear to apply to final image. Simulates vibrational modes.
+        :param sigma:       Convolve with gaussian of size sigma.
+        :param pupil_path:  If a path is given, load up a pupil image, else
+                            make image from galsim parameters.
+        """
+
+        # catch any kwargs passed along...
+        self.kwargs = kwargs
+
+        self.lam = lam
+        self.pupil_path = pupil_path
+        OpticalPSF = {'diam': 4.274419,
```

Comment:
This needs to be a parameter.  Likewise for things like nstruts, obstruction, etc. that you have hardcoded for DES.  Piff functionality shouldn't be hard-coded to DES values.

Although fine to have a parameter like `optical_template = 'des'`, which would populate DES values into parameters appropriately.  Then down the line we could have other templates for other telescopes.